lib/JsonpErrorHandlerPlugin.js

```diff
@@ -0,0 +1,295 @@
+			var jsonpFunction = this.outputOptions.jsonpFunction || Template.toIdentifier("webpackJsonp" + (this.outputOptions.library || ""));
+			var filename = this.outputOptions.filename || "bundle.js";
+			var chunkFilename = this.outputOptions.chunkFilename || "[id]." + filename;
+			var chunkMaps = chunk.getChunkMaps();
+			return this.asString([
+				source,
+				"",
+				"// install a JSONP callback for chunk loading",
+				"var parentJsonpFunction = window[" + JSON.stringify(jsonpFunction) + "];",
+				"window[" + JSON.stringify(jsonpFunction) + "] = function webpackJsonpCallback(chunkIds, moreModules) {",
+				this.indent([
+					'// add "moreModules" to the modules object,',
+					'// then flag all "chunkIds" as loaded and fire callback',
+					"var moduleId, chunkId, i = 0, callbacks = [];",
+					"for(;i < chunkIds.length; i++) {",
+					this.indent([
+						"chunkId = chunkIds[i];",
+						"if(installedChunks[chunkId])",
+						this.indent("callbacks.push.apply(callbacks, installedChunks[chunkId]);"),
+						"installedChunks[chunkId] = 0;"
+					]),
+					"}",
+					"for(moduleId in moreModules) {",
+					this.indent(this.renderAddModule(hash, chunk, "moduleId", "moreModules[moduleId]")),
+					"}",
+					"if(parentJsonpFunction) parentJsonpFunction(chunkIds, moreModules);",
+					"while(callbacks.length)",
+					this.indent("callbacks.shift().success.call(null, " + this.requireFn + ");"),
+					(this.entryPointInChildren(chunk) ? [
+						"if(moreModules[0]) {",
+						this.indent([
+							"installedModules[0] = 0;",
+							"return " + this.requireFn + "(0);"
+						]),
+						"}"
+					] : "")
+				]),
+				"};",
+				"",
+				"// load chunk",
+				"function loadChunk(chunkId) {",
+				this.indent([
+					"var head = document.getElementsByTagName('head')[0];",
+					"var script = document.createElement('script');",
+					"script.type = 'text/javascript';",
+					"script.charset = 'utf-8';",
+					"script.async = true;",
+					"function onComplete(error) {",
+						this.indent([
+							"// avoid mem leaks in IE.",
+							"script.onerror = script.onload = script.onreadystatechange = null;",
+							"if (error) {",
+							this.indent([
+								"var callbacks = installedChunks[chunkId];",
+								"// set chunkId to undefined so subsequent require's try again",
+								"delete installedChunks[chunkId];",
+								"if (callbacks) {",
+								this.indent([
+									"while(callbacks.length) {",
+										this.indent([
+											"callbacks.shift().error.call(null, __webpack_require__);",
+										]),
+									"}"
+								]),
+								'}',
+							]),
+							'}',
+							"// success callbacks will be called by webpackJsonpCallback handler"
+						]),
+					"}",
+					"script.onerror = script.onload = script.onreadystatechange = function() {",
+					this.indent([
+						"// cover buggy onerror / readystate implementations by checking whether the chunk is actually installed",
+						"onComplete(installedChunks[chunkId] !== 0);"
```

Comment:
To avoid buggy implementations of `onreadystatechange` in old IE (I think less than 9). It worked out nicely because webpack is already using a jsonp style callback so I just check whether it is actually installed to determine if an error has occurred.